plugin/trino-oracle/src/main/java/io/trino/plugin/oracle/OracleClient.java

```diff
@@ -351,7 +353,18 @@ else if (precision > Decimals.MAX_PRECISION || actualPrecision <= 0) {
 
             // This mapping covers both DATE and TIMESTAMP, as Oracle's DATE has second precision.
             case OracleTypes.TIMESTAMP:
-                return Optional.of(oracleTimestampColumnMapping());
+                if ("DATE".equalsIgnoreCase(typeHandle.getJdbcTypeName().orElseThrow(() -> new IllegalStateException("type name not present")))) {
```

Comment:
Usually, the typename-based mappins are done before the type number-based ones.
see https://github.com/trinodb/trino/blob/92e5ba6c03006bc306f7aff29c09c79d51154b57/plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java#L453-L469

Reply:
Also, can we use `typeHandle.getRequiredDecimalDigits()` to discern between Oracle's "date" and "timestamp"?